src/Api/PubnubApi/NewtonsoftJsonDotNet.cs

```diff
@@ -44,7 +45,7 @@ public NewtonsoftJsonDotNet(PNConfiguration pubnubConfig, IPubnubLog log)
         {
             this.config = pubnubConfig;
             this.pubnubLog = log;
-            JsonConvert.DefaultSettings = () => new JsonSerializerSettings { MaxDepth = 64 };
+            defaultJsonSerializerSettings = new JsonSerializerSettings { MaxDepth = 64 };
```

Comment:
Looks good. Instead of specify JsonConverter globally we are providing default behaviour only in scope of SDK.

Reply:
Exactly! That's the idea of this fix.